dask/bag/random.py

```diff
@@ -0,0 +1,123 @@
+import random
+
+
+def sample(population, k):
+    """
+    Chooses k unique random elements from a population bag.
+
+    Returns a new bag containing elements from the population while
+    leaving the original population unchanged.
+
+    Parameters
+    ----------
+    population : Bag
+        Elements to sample.
+    k : integer, optional
+        Number of elements to sample.
+    Examples
+    --------
+    >>> import dask.bag as db
+    >>> from dask.bag import random
+    >>>
+    >>> b = db.from_sequence(range(5), npartitions=2)
+    >>> len(list(random.sample(b, 3).compute()))
+    3
+    """
+    return _sample(population, k, False)
+
+
+def choices(population, k=1):
+    """
+    Return a k sized list of population elements chosen with replacement.
+
+    If the relative weights or cumulative weights are not specified,
+    the selections are made with equal probability.
+
+    Parameters
+    ----------
+    population : Bag
+        Elements to sample.
+    k : integer, optional
+        Number of elements to sample.
+    Examples
+    --------
+    >>> import dask.bag as db
+    >>> from dask.bag import random
+    >>>
+    >>> b = db.from_sequence(range(5), npartitions=2)
+    >>> len(list(random.choices(b, 3).compute()))
+    3
+    """
+    return _sample(population, k, True)
+
+
+def _sample(population, k=1, replace=False):
+    return population.map_partitions(_sample_map_partitions, k, replace).reduction(
+        lambda x: x, _sample_reduce
```

Comment:
I think you can probably just use `reduction` here and replace `lambda x: x` with `partial(_sample_map_partitions, k=k, replace=replace)` OR (even better) you can change `reduction` to accept `kwargs` like the one in `dask.dataframe` does:
https://github.com/dask/dask/blob/8efdf989cf44a03ac265ffde4342ec2348937902/dask/dataframe/core.py#L833-L835

Reply:
Hey really thanks, for the suggestion, using partial helped me out to simplify the overall algorithm.